### ISSUE TITLE
torch_mask 192 vs 384

### ISSUE THREAD

**Issue by ilkarman** [ElementAI/HighRes-net#2]
When the current patch-size of 64 is submitted to get_crop_mask from config the resultant tensor is of W,H: 192, 192 which is half the high-res image of (hr_maps) 384, 384 and causes a mismatch error at:

`cropped_mask = torch_mask[0] * hr_maps  # Compute `

> The size of tensor a (192) must match the size of tensor b (384)

Is this related to the following from the comp:

> To compensate for pixel-shifts, the submitted images are cropped by a 3 pixel border, resulting in a 378x378 format. These cropped images are then evaluated at the corresponding patches around the center of the ground-truth images, with the highest cPSNR being the score. In the following, HR is the ground-truth image and SR the submitted image, both in 384x384 resolution. We denote the cropped 378x378 images as follows: for all u,v∈{**0,…,6**}, HRu,v is the subimage of HR with its upper left corner at coordinates (u,v) and its lower right corner at (378+u,378+v).

In this case, wouldn't the get_crop_mask have this line

`mask = np.ones((1, 1, 6 * patch_size, 6 * patch_size))  # crop_mask for loss (B, C, W, H)
`
And multiply by 6 not 3? This runs but I'm not sure if it's correct

**Comment by MichelDeudon** [comment]
@ilkarman, if patch-size=64 in the config file, hr_maps should be of size (3*64,3*64) ie (192, 192). The hr_maps are cropped in DataLoader.py (see line 135 in read_imageset).
So the get_crop_mask() function should normally be fine. 

If that modification runs, it means you have super resolved images of size (384,384) in line 174 (srs) and 180 (shifted_srs) in train.py. Therefore your low-res images (lrs) are of size (128,128)? In order words, you are specifying a patch_size but you are not creating patches. 

I found the error! It's in config/config.json. You need to specify create_patches=true.
Will resolve this issue.

Thank you!

**Comment by ilkarman** [comment]
Thanks a lot, runs perfect now!

**Comment by MichelDeudon** [comment]
Awesome!
